lupa-core/src/main/kotlin/org/jetbrains/research/lupa/kotlinAnalysis/util/JupyterInverter.kt

```diff
@@ -0,0 +1,144 @@
+package org.jetbrains.research.lupa
+
+import org.json.JSONArray
+import org.json.JSONObject
+import java.io.File
+import java.io.IOException
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.util.*
+
+class JupyterInverter(input: String, output: String) {
+    val inputDir = input
+    val outputDir = output
+
+    class Notebook(notebookJSON: JSONObject, filename: String) {
+        var markdownCells: JSONArray
+        var codeCells: JSONArray
+        val metaData: JSONObject = notebookJSON.get("metadata") as JSONObject
+        val ntbName: String = filename
+
+        init {
+            this.markdownCells = getMarkdownCells(notebookJSON)
+            this.codeCells = getCodeCells(notebookJSON)
+        }
+
+        private fun cellToArray(cellJSON: JSONObject): JSONArray {
+            val source = cellJSON.get("source")
+            return if (source is String) {
+                val arr = source.split("\n").toMutableList()
+                for (i in 0 until arr.size - 1) {
+                    arr[i] = arr[i] + "\n"
+                }
```

Comment:
What are you trying to do? In Kotlin are exist a lot of functions to work with collections, please, avoid using for 
Also this function needs in comments since it is unclear...